### ISSUE TITLE
data param in transformRequest is undefined

### ISSUE THREAD

**Issue by JuoCode** [mzabriskie/axios#161]
I want add API endpoint in transformRequest, but data param is undefined :joy:
```
const API = axios.create({
  transformRequest: [(data, headers) => {
    console.dir(data);
    console.dir(headers);
    return data;
  }]
});
```

I use master

**Comment by nickuraltsev** [comment]
@JuoCode Can you share the code that sends the request?

**Comment by JuoCode** [comment]
```
export function fetchProjects () {
  return dispatch => {
    return API.get('http://example.com/projects')
      .then(response => {
        dispatch({ type: PROJECTS_RECEIVED, payload: response.data });
      });
  };
}
```

**Comment by nickuraltsev** [comment]
`transformRequest` allows changing the request body (`data`) and headers. Your code makes a GET request that has no body. That's why `data` is undefined. 

If can try to use interceptors to change other request properties.

```js
API.interceptors.request.use(
  config => {
    // request config can be changed here
    return config;
  }, 
  error => Promise.reject(error)
);
```

**Comment by JuoCode** [comment]
Get it! #160 is necessary

**Comment by mzabriskie** [comment]
Just released [0.8.0](https://github.com/mzabriskie/axios/blob/master/CHANGELOG.md#080-dec-11-2015) which includes #160.
